### ISSUE TITLE
Travis fails to deploy _site folder to gh-pages

### ISSUE THREAD

**Issue by saifkhichi96** [saifkhichi96/saifkhichi96.github.io#10]
I am using the `jekyll-paginate-v2` plugin, which isn't supported by GitHub Pages. I don't want to build the site manually every time I write a new blog post or make an update, so I am trying to use Travis CI to automatically build my site and then push the static build files (output to `_site` directory by default) to the `gh-pages` branch.

My `.travis.yml` looks like this:
```
language: ruby
rvm: 2.4.1

branches:
  only:
    - master

notifications:
  email:
    on_success: always
    on_failure : always

script:
 - gem update --system
 - gem install bundler
 - bundler update --bundler
 - gem install jekyll bundler
 - bundle exec jekyll build

deploy:
  provider: pages
  local_dir: _site/
  token: $TRAVIS_DEPLOY_TOKEN
  keep_history: true
  target_branch: gh-pages
  fqdn: www.saifkhichi.com
  on:
    branch: master

env:
  global:
  - NOKOGIRI_USE_SYSTEM_LIBRARIES=true # speeds up installation of html-proofer
```

And as can be seen in the [build log](https://travis-ci.com/github/saifkhichi96/saifkhichi96.github.io/jobs/487413364/), the build completes with destination set to `_site` folder as expected.
```
Configuration file: /home/travis/build/saifkhichi96/saifkhichi96.github.io/_config.yml
            Source: /home/travis/build/saifkhichi96/saifkhichi96.github.io
       Destination: /home/travis/build/saifkhichi96/saifkhichi96.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating... 
       Jekyll Feed: Generating feed for posts
         AutoPages: Disabled/Not configured in site.config.
        Pagination: Complete, processed 1 pagination page(s)
                    done in 0.607 seconds.
 Auto-regeneration: disabled. Use --watch to enable.
The command "bundle exec jekyll build" exited with 0.
```

However, the deployment fails with the following message. It says that it could not copy the `_site` folder to deploy destination. My `TRAVIS_DEPLOY_TOKEN` is set correctly, which I have verified by deploying the root folder instead of `_site`, which completes successfully.

```
$ rvm $(travis_internal_ruby) --fuzzy do ruby -S gem install dpl
Installing deploy dependencies
Logged in as @saifkhichi96 (Muhammad Saif Ullah Khan)
Preparing deploy
Deploying application
Already up to date!
HEAD detached at b6ae57a
nothing to commit, working tree clean
Dropped refs/stash@{0} (c6ba0606e54e2a05b6fc4997d9bbdc99a5eb49c2)
Could not copy /home/travis/build/saifkhichi96/saifkhichi96.github.io/_site.
failed to deploy
```

**Comment by saifkhichi96** [comment]
Fixed with 06253b16feceecdd77b8cedb89ec8aa19032e6bc

**Comment by saifkhichi96** [comment]
I was missing the `skip_cleanup: true` under `deploy` from the `.travis.yml`. By default, Travis deletes all the build outputs, which is why it was failing to push those files to GitHub. Skipping cleanup stops the files from being deleted before they can be deployed.
